### PR TITLE
#22368 Fix identifiers case sensivity in semantical validation

### LINKED ISSUE #22368: Syntax coloring indicates error on object in lower case even without double-quote [SNOWFLAKE]
### Description

In Snowflake DB, name of DB objects (schema, tables, fields etc.) are by default interpreted as uppercased from the SQL Client, unless (and only unless) the name is explicitely enclosed by double quote.
For instance, the standard schema `INFORMATION_SCHEMA` and its standard view `TABLES` can be either reached out to via calling them `INFORMATION_SCHEMA.TABLES`, `information_schema.tables`, `inForMatioN_SChema.TablEs` etc. The SF engine will always look for `INFORMATION_SCHEMA.TABLES`.

Only if you use double-quotes, then only the case is taken into consideration : `INFORMATION_SCHEMA` <> `"information_schema"`.

However, the newly added auto-coloring feat in DBeaver, indicating potential errors in syntax, takes the case in consideration even without double-quotes...

image
 vs
 
image

### DBeaver Version

CE Version 23.3.2.202401072024

### Operating System

Win11 23H2 (Build 22631.2715)

### Database and driver

Snowflake via JDBC 3.13.29

### Steps to reproduce

(see screenshots provided)

### Additional context

_No response_

### LINKED ISSUE #22385: semantic analysis doesn't work as intended
### Description

Everything uppercase
image

Library lowercase
image

"Read database table/columns for semantic analysis" turned off
image

With the setting "Enable semantic analysis (experimental)" on, I have to write everything in capital letters, otherwise it marks tables and columns as errors. 
Turning off "Read database table/columns for semantic analysis" still marks columns as Errors. 
I use DB2 for IBM i connection.
I have no problems with mysql, regardless of whether I use upper or lower case.

Didn't have this issue with 23.3.1 

### DBeaver Version

23.3.2

### Operating System

Windows 10

### Database and driver

image

### Steps to reproduce

1. Open a DB2 for IBM i Connection 
2. Write any SELECT statement with everything uppercase --> no problems
3. Change library or table to lowercase  --> marked as false but executable

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by E1izabeth**
### ONE PR FOR TWO ISSUES #22368 #22385

**For QA**
Please check case sensitivity for identifiers (including quoted identifiers) in various databases (DB2, Snowflake, SQL Server, PostgreSQL, SQLite, MySQL, Oracle, Clickhouse)

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLSemanticsUtils.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ui.editors.sql.semantics;
+
+import org.jkiss.dbeaver.model.sql.SQLDialect;
+
+public class SQLSemanticsUtils {
+
+    public static String identifierToCanonicalForm(SQLDialect dialect, String rawIdentifierString, boolean forceUnquotted) {
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLSemanticsUtils.java]: Please add NotNull annotations

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLSemanticsUtils.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ui.editors.sql.semantics;
+
+import org.jkiss.dbeaver.model.sql.SQLDialect;
+
+public class SQLSemanticsUtils {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLSemanticsUtils.java]: Why adding new utility class?
We have SQLUtils and SQLSemanticProcessor.
Maybe we need to refactor some utils but adding new utils with just 1 function looks odd.

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Review by LonwoLonwo — APPROVED** [review]

**Comment by Matvey16** [comment]
Now correct column names in quotes are highlighted as incorrect

https://github.com/dbeaver/dbeaver/assets/82543000/e9dafb6d-fd9d-43c6-bf0d-9ce405ed618a

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLUtils.java [hunk 3]
```diff
@@ -1158,4 +1159,26 @@ public static String getDefaultScriptDelimiter(SQLDialect sqlDialect) {
...
+    /**
+     * Returns identifier in canonical form depends on dialect parameters. It's useful when we need to compare two identifiers.
+     */
+    public static String identifierToCanonicalForm(
+        @NotNull SQLDialect dialect,
+        @NotNull String rawIdentifierString,
+        boolean forceUnquotted,
+        boolean prepared
+    ) {
+      String actualIdentifierString;
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLUtils.java]: Indentation

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Comment by Matvey16** [comment]
Verified

**Review by serge-rider — APPROVED** [review]

**Pull request merged**